lib/qless/worker/base.rb

```diff
@@ -50,11 +50,17 @@ def initialize(reserver, options = {})
       # USR1: Kill the forked children immediately, continue processing jobs.
       # USR2: Pause after this job
       # CONT: Start processing jobs again after a USR2
+      #  HUP: Print current stack to log and continue
       def register_signal_handlers
         # Otherwise, we want to take the appropriate action
         trap('TERM') { exit! }
         trap('INT')  { exit! }
         begin
+          trap('HUP') { log_stack_trace }
+        rescue ArgumentError
```

Comment:
I do not know enough regarding support, and decided to play it safe. After all, if SIGHUP is not supported the rest of functionality do not suffer.

Regarding extraction of traps, I thought a bit about it. There is a good reason to trap USR2 & CONT together -- otherwise one might pause, and unable to unpause worker(s). OTOH, trap does not work as transaction, so we might want to separate USR1, and swap order of USR2/CONT. Safe_trap sound good, I can implement it.